src/Microsoft.Compatibility.HashCode/ref/Configurations.props

```diff
@@ -0,0 +1,14 @@
+<Project>
+  <PropertyGroup>
+    <PackageConfigurations>
+      netstandard;
+      netstandard2.1;
+      netcoreapp2.1;
+      uap;
```

Comment:
Weren't we saying that netstandard2.1 ought to be good enough to not include uap/netcoreapp builds?  I guess you need the netcoreapp build for source build so that it doesn't get the `netcoreapp2.1` config.

Reply:
> I guess you need the netcoreapp build for source build 

Do we need to build this in source build?

Ideally, these packages would live in separate repo, ship on its own schedule and not be attached to .NET Core engineering system at all - but I understand that we are worried about the cost of maintaining a new repo.

Reply:
Technically no, it's not needed in source build.  You could avoid the netcoreapp configuration entirely and just condition out the project.

I don't think packages that shared the majority of their sources with the product make sense to move to a seperate repo, the seperate repo will be the servicing branch for 3.0.

Reply:
I look at these packages as separate product from .NET Core 3.0. The original proposal was that they will built in separate repo, ship some time after .NET Core 3.0 ships, etc.

Yes, there is a code duplication - but this code duplication is good and makes things simpler to reason about.

Reply:
I guess you predict the future a bit differently than me.  I'm imagining that until we ship this stuff we're going to want to make any fixes in the same place without having to remember to copy it to another repository. We're going to want the same test cases to run in both places so that we ensure the downlevel stuff has the same behavior as the inbox stuff.  Once we ship it out of 3.0 and maintain that release we want to make sure that any security/reliability fixes happen to the same codebase, so that we don't need to remember to copy those to another repo.

I agree with your sentiment about not wanting to burden mainline with having to maintain the downlevel packages, but I think I'd handle that by letting it spin off in servicing rather than bringing up an entire new repo and invent a bunch of new engineering process for the sake of simplifying a single release.

On the otherhand, if you envisioning the compat packs needing to be maintained for a long time independently of the 3.0 release such that we couldn't justify making the sort of fixes we'd need in a servicing branch I could be persuaded otherwise, but I believe that's actually an anti-goal here.

Reply:
> the servicing branch for 3.0

Do we understand servicing SLA for these packages? Is the servicing branch for 3.0 going to live long-enough for these packages?

Reply:
Replace 3.0 with "whatever becomes the LTS for 3.x".  That may mean we wait to delete the packages/configuration for a little bit if we decide we don't want to make 3.0 LTS.  If we don't make 3.0 LTS then I suspect that codebase wouldn't be hindered too much by having to maintain these configurations until we have an LTS.

Reply:
> Weren't we saying that netstandard2.1 ought to be good enough to not include uap/netcoreapp builds?

netcoreapp2.1 has to be there because it doesn't support netstandard2.1 and HashCode is inbox on netcoreapp2.1. The netcoreapp versionless configuration is needed in order to be able to build vertically. Uap configurations may be deleted.

Reply:
> whatever becomes the LTS for 3.x

Is LTS for 3.x going to have long enough life for these packages?

These packages are set in stone. Their public surface cannot be reasonably evolved. I expect that we will stop building these packages in 4.x or 5.x master because of it will conflict with evolution of the code, just like it happened for many existing packages that mirrored inbox components.

Reply:
> Is LTS for 3.x going to have long enough life for these packages?

I presume so.  We already do the same in 2.1 for System.Memory, System.Threading.Tasks.Extensions, others.  We can vet this with our release managers & folks defining the SLA.

Reply:
Fixed the main comment here.